File: MyMindNotes/app/src/main/java/com/android/mymindnotes/model/retrofit/GetDiaryListApi.java
package com.android.mymindnotes.model.retrofit;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface GetDiaryListApi {
    @GET("/api/diary/getAll/{user_index}")
    Call<Map<String, Object>> getAllDiary(@Path("user_index") int user_index);
}

File: MyMindNotes/app/src/main/java/com/android/mymindnotes/data/retrofit/api/user/GetDiaryListApi.kt
package com.android.mymindnotes.data.retrofit.api.user

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface GetDiaryListApi {
    @GET("/api/diary/getAll/{user_index}")
    suspend fun getAllDiary(
        @Path("user_index") user_index: Int
    ): Map<String, Object>
}